web/src/App.tsx

```diff
@@ -37,16 +37,23 @@ function Theme() {
             return themes.Dark;
         case "grey":
             return themes.Grey;
+        case "auto":
+            return window.matchMedia("(prefers-color-scheme: dark)").matches ? themes.Dark : themes.Light;
         default:
             return themes.Light;
     }
 }
 
 const App: React.FC = () => {
     const [notification, setNotification] = useState(null as Notification | null);
-
+    const [theme, setTheme] = useState(Theme());
+    if (getTheme() === "auto") {
+        window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", (e) => {
```

Comment:
The listener should be a callback created with useCallback and setTheme should be a dependency of this callback I think.

Reply:
I'm very new to React so I might be wrong, but it seems as if the `useCallback` hook will call a function if a dependency is changed. In this case, the value used is not a stateful dependency, so I don't think `useCallback` would be applicable.

Reply:
The component function will be executed everytime the component needs to be rendered (which is out of your control) and in general in React you want to control when the callbacks are called so in general the calls should not be in the rendered section at all but rather in some kind of callback. In your case, you're right, you want to execute setTheme only once (no dep change) so I was wrong, you need a useEffect instead of useCallback with setTheme in the dependency list.

Reply:
Thanks for the clarification, I see the issue now.